main.go

```diff
@@ -188,11 +271,12 @@ func getQuestionKey(q dns.Question) string {
 	return n + t + c
 }
 
-func getMsgKey(r *dns.Msg) string {
+func getMsgKey(r *dns.Msg, limit int) string {
```

Comment:
would it be possible to move determining the limit value into here (since the msg is here anyway), that way it wouldn't be necessary to pass the limit around everywhere?